src/link/MachO/load_commands.zig

```diff
@@ -467,8 +467,27 @@ pub inline fn appleVersionToSemanticVersion(version: u32) std.SemanticVersion {
     };
 }
 
-pub fn inferSdkVersionFromSdkPath(path: []const u8) ?std.SemanticVersion {
-    const stem = std.fs.path.stem(path);
+pub fn inferSdkVersion(gpa: Allocator, comp: *const Compilation) ?std.SemanticVersion {
+    var arena_allocator = std.heap.ArenaAllocator.init(gpa);
+    defer arena_allocator.deinit();
+    const arena = arena_allocator.allocator();
+
+    const options = comp.bin_file.options;
+
+    const sdk_layout = options.darwin_sdk_layout.?;
```

Comment:
Same here. Unless we are absolutely certain `darwin_sdk_layout != null` at all times, we should instead return a `null` here if the former is `null` and let the linker carry on with linking attempt and report any errors. Something like the following:

```zig
const sdk_layout = options.darwin_sdk_layout orelse return null;
```